crates/solver/src/settlement.rs

```diff
@@ -242,7 +242,11 @@ pub struct Settlement {
     pub encoder: SettlementEncoder,
     pub submitter: SubmissionPreference, /* todo - extract submitter and score into a separate
                                           * struct */
+    /// TODO remove this field once all solvers conform to sending the
+    /// success_probability
     pub score: Option<Score>,
+    pub success_probability: Option<f64>,
+    pub gas_amount: Option<U256>,
```

Comment:
I'm a bit unhappy we are leaking these new fields to a lot of otherwise unrelated data models (adding these two fields here makes the otherwise relatively clean settlement struct quite messy).

Is there a reason to not put these two values as another arm on `Score` itself? Something like
```
pub enum Score {
    Solver(U256),
    Discount(U256),
    Risk(f64,Option<U256>)
}
```

We don't allow solvers to specify a mix of the two. When we remove the old score, we can just flatten this field to `RiskAdjustment`